payroll/models/hr_payslip.py

```diff
@@ -587,117 +466,127 @@ def get_contract_dict(self, contract, contracts):
 
         return {}
 
-    @api.model
-    def _get_payslip_lines(self, contract_ids, payslip_id):
-        def _sum_salary_rule_category(localdict, category, amount):
-            if category.parent_id:
-                localdict = _sum_salary_rule_category(
-                    localdict, category.parent_id, amount
-                )
-
-            if category.code in localdict["categories"].dict:
-                localdict["categories"].dict[category.code] += amount
-            else:
-                localdict["categories"].dict[category.code] = amount
-
-            return localdict
-
-        payslip = self.env["hr.payslip"].browse(payslip_id)
-        contracts = self.env["hr.contract"].browse(contract_ids)
-
-        # we keep a dict with the result because a value can be overwritten by
-        # another rule with the same code
-        result_dict = {}
-        rules_dict = {}
-        contract_dict = {}
-        blacklist = []
-        categories = BrowsableObject(payslip.employee_id.id, {}, self.env)
-        rules = BrowsableObject(payslip.employee_id.id, rules_dict, self.env)
-        current_contract = BrowsableObject(
-            payslip.employee_id.id, contract_dict, self.env
-        )
-
-        baselocaldict = payslip.get_baselocaldict(contracts)
-        baselocaldict["categories"] = categories
-        baselocaldict["rules"] = rules
-        baselocaldict["current_contract"] = current_contract
+    def _get_baselocaldict(self, contracts):
+        self.ensure_one()
+        worked_days_dict = {
+            line.code: line for line in self.worked_days_line_ids if line.code
+        }
+        input_lines_dict = {
+            line.code: line for line in self.input_line_ids if line.code
+        }
+        localdict = {
+            **{
```

Comment:
Sorry, I don't understand why this is necessary.

Reply:
This is useful if you want to merge the dicts and be available as keyword arguments. It's similar to **kargs in functions.

Reply:
It can be removed, I added it because it makes no change an allow to inject a whole dict easily if needed but this yes I don't use it for now. 

But if not used it makes no difference.